src/expr.rs

```diff
@@ -2546,3 +2566,29 @@ fn rewrite_expr_addrof(
     };
     rewrite_unary_prefix(context, operator_str, expr, shape)
 }
+
+pub trait ExprTypeIdentifier {
+    fn is_type(&self) -> bool;
+    fn is_expr(&self) -> bool {
+        !self.is_type()
+    }
```

Comment:
Why not implement is_expr directly and skip is_type? But see my previous comment for perhaps some way to eliminate is_expr altogether